utils/compare_generated.sh

```diff
@@ -1,17 +1,19 @@
 #!/bin/bash
-if [ "$#" -lt 2 ];
+
+if [ "$#" -lt 3 ];
```

Comment:
I wouldn't mind a check for `3 <= $# <= 4`, but I'll let this slide. The more common case is "I don't know how to run it so here let me pass 0 arguments or a `-h` / `--help` to figure out how and this catches that, so I'm inclined to leave it. More defensive coding would validate that args 1 and 2 are directories that exist, args 3 and 4 are the correct strings if they exist, but I guess that's unnecessary since it is a manually run script that doesn't generate any build/etc. artifacts.